configure.ac

```diff
@@ -155,6 +155,35 @@ define_xopen_source=yes
 AC_SUBST(CONFIG_ARGS)
 CONFIG_ARGS="$ac_configure_args"
 
+dnl Allow users to disable pkg-config or require pkg-config
+AC_ARG_WITH(
+  [pkg-config],
+  [AS_HELP_STRING([--with-pkg-config], [use pkg-config to detect build options (default is check)])],
+  [],
+  [with_pkg_config=check]
+)
+AS_CASE([$with_pkg_config],
+  [yes|check], [
+    if test -z "$PKG_CONFIG"; then
+      dnl invalidate stale config.cache values
+      AS_UNSET([PKG_CONFIG])
+      AS_UNSET([ac_cv_path_ac_pt_PKG_CONFIG])
+      AS_UNSET([ac_cv_prog_ac_ct_PKG_CONFIG])
+    fi
```

Comment:
Unsetting `PKG_CONFIG` and friends if `PKG_CONFIG` is not set? Is there a missing `!` in the test? Is this check needed at all; the config.cache guard is pretty strict.

Reply:
It's un-declaring the variables when it's empty. This solves a corner case when going from ``./configure -C --without-pkg-config`` to ``./configure -C --with-pkg-config=yes``.

Reply:
Maybe you are able to find a better way...

Reply:
I see. No, this is fine. I can't think of a better way.